### PR TITLE
Include explicit separator when constructing path to Windows global config file

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/9416

### CODE REVIEW

**Comment by zanieb** [comment]
😭 

https://github.com/astral-sh/uv/blob/9fd6958a16f56c1e3fd4d40cb82df41ff1737e60/crates/uv-settings/src/lib.rs#L369-L378

File: crates/uv-settings/src/lib.rs [hunk 1]
```diff
@@ -201,10 +201,25 @@ fn locate_system_config_xdg(value: Option<&str>) -> Option<PathBuf> {
...
+    //
+    // See https://github.com/rust-lang/rust/blob/7db7489f9bc274cb60c4956bfa56de0185eb1b9b/library/std/src/path.rs#L1305-L1313
+    let comps = candidate.components();
+    let prefix = if comps.prefix_len() > 0
+        && comps.prefix_len() == comps.path.len()
+        && comps.prefix.unwrap().is_drive() {
+            "\\"
+        } else {
+            ""
+        };
```
Comment (zanieb) [inline crates/uv-settings/src/lib.rs:220]: So... `prefix` is private. I'm not sure what to do here. I think we need to change the way we're testing this?

**Comment by charliermarsh** [comment]
Taking a look on my Windows machine.

**Comment by zanieb** [comment]
Replaced by https://github.com/astral-sh/uv/pull/9488

**Pull request closed without merging**